src/pages/Add.tsx

```diff
@@ -101,59 +74,70 @@ const Add = () => {
     return target.length === length ? returnValue : '';
   };
 
-  const checkCardNumberPasswordValidation = () => {
-    const copyCardNumberPasswordError = [...cardNumberPasswordError];
-
-    if (isNaN(Number(cardNumber[2]))) copyCardNumberPasswordError[0] = true;
-    else copyCardNumberPasswordError[0] = false;
+  const checkCardNumberValidation = () => {
+    const copyCardNumbers = [...formErrors.cardNumbers];
 
-    if (isNaN(Number(cardNumber[3]))) copyCardNumberPasswordError[1] = true;
-    else copyCardNumberPasswordError[1] = false;
+    cardNumber.forEach((number, index) => {
+      if (isValidNumber(number)) copyCardNumbers[index] = true;
+      else copyCardNumbers[index] = false;
+    });
 
-    setCardNumberPasswordError([...copyCardNumberPasswordError]);
+    setFormErrors({ ...formErrors, cardNumbers: [...copyCardNumbers] });
   };
 
   const checkExpirationNumberValidation = () => {
     const mm = Number(expirationNumber[0]);
 
-    if (mm >= 1 && mm <= 12) setExpirationMonthError(false);
-    else setExpirationMonthError(true);
+    if (mm >= 1 && mm <= 12)
```

Comment:
분기문 없이 이렇게 실행해도 괜찮을 것 같네요 
```suggestion
    const isValid = mm >= 1 && mm <= 12; 
    setFormErrors({ ...formErrors, expirationMonth: isValid });
```

Reply:
반영했습니다~! [756e737](https://github.com/next-step/react-payments/pull/25/commits/756e7375316afe605d202830b8092d417e3fab93)